src/main.rs

```diff
@@ -176,33 +184,45 @@ fn run_shell(vm: &mut VirtualMachine) -> PyResult {
         println!("No previous history.");
     }
 
-    let ps1 = &objstr::get_value(&vm.sys_module.get_attr("ps1").unwrap());
-    let ps2 = &objstr::get_value(&vm.sys_module.get_attr("ps2").unwrap());
-    let mut prompt = ps1;
+    let mut prompt = get_prompt(vm, "ps1");
+
+    let mut continuing = false;
 
     loop {
-        match repl.readline(prompt) {
+        if !continuing {
```

Comment:
I like the mutable boolean continueing, but I would like to change the prompt into an immutable like this:
```rust
let prompt = if continuing {
  get_prompt(vm, "ps2")
} else {
  get_prompt(vm, "ps1")
};
```
This way, prompt is only set once.

Reply:
Ah, thanks for noticing that it could be refactored in that way.